hoot-core/src/main/cpp/hoot/core/io/HootNetworkRequest.cpp

```diff
@@ -150,19 +156,48 @@ bool HootNetworkRequest::_networkRequest(const QUrl& url, int timeout,
     return false;
     break;
   }
-  //  Wait for finished signal from reply object
-  _blockOnReply(reply, timeout);
-  //  Get the status and content of the reply if available
-  _status = _getHttpResponseCode(reply);
-  //  According to the documention this shouldn't ever happen
-  if (reply == NULL)
+  //  Start the timer
+  timeoutTimer.start(timeout * 1000);
+  //  Connect the timeout lambda
+  QObject::connect(&timeoutTimer, &QTimer::timeout, [&]()
+  {
+    //  Stop the timer first so the lambda isn't called again
+    timeoutTimer.stop();
+    //  Set the timed out flag before aborting the reply
+    _timedOut = true;
+    //  Abort will trigger QNetworkAccessManager::finished()
+    reply->abort();
+    //  Set the status to negative TimeoutError, negative so that it doesn't
+    //  collide with HTTP status numbers
+    _status = -1 * (int)QNetworkReply::TimeoutError;
+    _error = QString("HTTP Request to %1 timed-out.").arg(tempUrl.toString(QUrl::RemoveUserInfo));
+    //  Exit with a non-zero status
+    loop.exit(_status);
+  });
```

Comment:
The timeout lambda stops the timer (because it repeats), aborts the request, and sets the error string to the new, easier to read error message.